tests/unit/yamlFilterTests.cpp

```diff
@@ -0,0 +1,233 @@
+#define CATCH_CONFIG_MAIN
+#include "catch.hpp"
+
+#include <iostream>
+#include <vector>
+
+#include "filters.h"
+#include "yaml-cpp/yaml.h"
+#include "sceneLoader.h"
+
+using namespace Tangram;
+using namespace YAML;
+
+SceneLoader sceneLoader;
+std::vector<Feature> vehicles;
+Context ctx;
+
+void init() {
+    vehicles.clear();
+    Feature civic;
+    civic.props.stringProps["name"] = "civic";
+    civic.props.stringProps["brand"] = "honda";
+    civic.props.numericProps["wheel"] = 4;
+    civic.props.stringProps["drive"] = "fwd";
+    civic.props.stringProps["type"] = "car";
+    civic.props.numericProps["fancy"] = 0;
+    vehicles.push_back(civic);
+
+    Feature bmw1;
+    bmw1.props.stringProps["name"] = "bmw320i";
+    bmw1.props.stringProps["brand"] = "bmw";
+    bmw1.props.stringProps["series"] = "3";
+    bmw1.props.numericProps["wheel"] = 4;
+    bmw1.props.stringProps["drive"] = "all";
+    bmw1.props.stringProps["type"] = "car";
+    bmw1.props.numericProps["fancy"] = 1;
+    vehicles.push_back(bmw1);
+
+    Feature bike;
+    bike.props.stringProps["name"] = "cb1100";
+    bike.props.stringProps["brand"] = "honda";
+    bike.props.numericProps["wheel"] = 2;
+    bike.props.stringProps["type"] = "bike";
+    bike.props.stringProps["series"] = "CB";
+    bike.props.numericProps["fancy"] = 1;
+    vehicles.push_back(bike);
+
+    vehicles.swap(vehicles);
+
+    ctx["$vroom"] = new NumValue(1);
+    ctx["$zooooom"] = new StrValue("false");
+}
+
+
+//1. basic predicate
+TEST_CASE( "yaml-filter-tests: basic predicate test", "[filters][core][yaml]") {
+    init();
+    YAML::Node node = YAML::Load("filter: { series: 3}");
+    Filter* filter = sceneLoader.generateFilter(node["filter"]);
+    int count = 0;
+    for(auto& vehicle : vehicles) {
+        if(filter->eval(vehicle, ctx)) {
+            count++;
+        }
+    }
+    REQUIRE(count == 1);
```

Comment:
I'm not sure counting is the best way to evaluate these tests. In this test the filter could pass the wrong 'vehicle' and the test wouldn't complain.